### ISSUE TITLE
Unexpected set_parallel behavior in SceneTreeTween

### ISSUE THREAD

**Issue by firebelley** [godotengine/godot#69264]
### Godot version

3.5.1.stable.mono

### System information

Windows 11

### Issue description

The `set_parallel` function of `SceneTreeTween` explicitly states that "Tweeners appended after this method will by default run simultaneously, as opposed to sequentially."

However, this is not the case. A tween defined before the call to `set_parallel` appears not to run at all. Or perhaps the tween defined after `set_parallel` is being run prematurely thus overriding it.

```
# Root node is a ColorRect
var tween = create_tween()
tween.tween_property(self, "color", Color.black, 1)
tween.tween_property(self, "color", Color.white, 1)
tween.set_parallel()
tween.tween_property(self, "color", Color.red, 1)
```
In the above snippet, I expect the following to happen:
1. Tween the `ColorRect` to black over 1 second.
2. Tween the `ColorRect` to white over  1 second.
3. Enable parallel execution
4. Tween the `ColorRect` to red over 1 second.

However, the observed behavior is instead as follows:
1. The `ColorRect` is tweened to black over 1 second.
2. The `ColorRect` is tweened to red over 1 second.

In the observed behavior, the transition back to white never happens. The tween goes immediately from black to red, despite the fact that a tween to white is specified _before_ `set_parallel` is called.

### Steps to reproduce

See issue description or minimal reproduction project.

### Minimal reproduction project

[SceneTreeTween Issue Example.zip](https://github.com/godotengine/godot/files/10099078/SceneTreeTween.Issue.Example.zip)

**Comment by KoBeWi** [comment]
`set_parallel()` has the same immediate effect as `parallel()`, i.e. the next Tweener will run in parallel to the previously defined one. In your case white and red animations happen at the same time and red is overwriting the color, because it's animated later.

**Comment by firebelley** [comment]
Yeah that doesn't make sense given the description. The description of `set_parallel` only mentions tweens set after the call. Arguably it shouldn't have an immediate effect, otherwise every tween defined before it is called will be parallelized which, I would guess, would be undesired in almost every case. At the very least it's not intuitive.

**Comment by Lippanon** [comment]
This seems to be the 3.x version of https://github.com/godotengine/godot/issues/74608. In 4.x the documentation was addressed by https://github.com/godotengine/godot/pull/79758.
